src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.ts

```diff
@@ -52,7 +63,40 @@ export class EditSingleEntityComponent
       (entity) => entity.getId() === this.formControl.value
     );
     if (selectedEntity) {
+      this.selectedEntity = selectedEntity;
+      this.editingSelectedEntity = false;
       this.entityNameFormControl.setValue(selectedEntity.toString());
+    } else {
+      this.entityNameFormControl.setValue("");
     }
   }
+
+  select(event: MatAutocompleteSelectedEvent) {
+    const entity = this.entities.find(
+      (e) => e.toString().toLowerCase() === event.option.value.toLowerCase()
+    );
+    if (entity) {
+      this.selectedEntity = entity;
+      this.editingSelectedEntity = false;
+      this.formControl.setValue(entity.getId());
+    }
+  }
+
+  submit() {
```

Comment:
This function is very similar to the `select` function. I guess the function parameter `event` could be changed to `entityString: string` and then it just has to be called with either `event.option.value` or `entityNameFormControl.value`.